### PR TITLE
feat(Member): add flags

### PR DESCRIPTION
**Pull request by Victorsitou**
## Summary
https://github.com/discord/discord-api-docs/pull/5874

## Checklist

- [ ] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/member.py [hunk 1]
```diff
@@ -427,6 +431,7 @@ def _update(self, data: GuildMemberUpdateEvent) -> None:
         self._avatar = data.get("avatar")
         timeout_datetime = utils.parse_time(data.get("communication_disabled_until"))
         self._communication_disabled_until = timeout_datetime
+        self._flags = data["flags"]
```
Comment (shiftinv) [inline disnake/member.py:434]: `_update_from_message` is missing this change

**Review by shiftinv — CHANGES_REQUESTED** [review]
Mostly documentation stuff, looks good to me overall.

It's worth mentioning that editing flags isn't documented (neither is the permission requirement); still, it works as expected and the client does it this way too.

File: disnake/flags.py [hunk 2]
```diff
@@ -2134,3 +2135,102 @@ def slurs(self):
         (contains insults or words that may be considered hate speech).
         """
         return 1 << 3
+
+
+class MemberFlags(BaseFlags):
+    """Wraps up Discord Member flags.
+
+    .. versionadded:: 2.8
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/flags.py]: a600e1bbfc6c23d2bc859e0610d29c90247eaf45

File: disnake/member.py [hunk 3]
```diff
@@ -816,6 +833,17 @@ async def edit(
 
             .. versionadded:: 2.3
 
+        flags: :class:`MemberFlags`
+            The member's new flags. To know what flags are editable,
+            see `the documentation <https://discord.com/developers/docs/resources/guild#guild-member-flags>`_.
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/member.py]: a600e1bbfc6c23d2bc859e0610d29c90247eaf45

[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/member.py]: a600e1bbfc6c23d2bc859e0610d29c90247eaf45

File: disnake/flags.py [hunk 4]
```diff
@@ -2134,3 +2135,102 @@ def slurs(self):
...
+
+    .. container:: operations
+
+        .. describe:: x == y
+            Checks if two MemberFlags instances are equal.
+
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/flags.py:2146]: a600e1bbfc6c23d2bc859e0610d29c90247eaf45

File: docs/api.rst [hunk 5]
```diff
@@ -6092,6 +6092,13 @@ PublicUserFlags
 .. autoclass:: PublicUserFlags()
     :members:
 
+MemberFlags
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply docs/api.rst:6095]: 162ded0d2a4022b04e9e900b6b9fcf016867d97a

File: disnake/member.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/member.py]: 9d08bf5df8d5916240de7f9a76ea299aa3c5d44b

File: disnake/member.py [hunk 6]
```diff
@@ -336,6 +338,7 @@ def __init__(
         self._avatar: Optional[str] = data.get("avatar")
         timeout_datetime = utils.parse_time(data.get("communication_disabled_until"))
         self._communication_disabled_until: Optional[datetime.datetime] = timeout_datetime
+        self._flags: int = data["flags"]
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/member.py]: 9d08bf5df8d5916240de7f9a76ea299aa3c5d44b

File: disnake/member.py (hunk 1 above)
Comment (Victorsitou) [reply disnake/member.py]: a68ddb62a99a4df3860fd97cb775a95f84714b65

File: disnake/member.py [hunk 7]
```diff
@@ -816,6 +833,17 @@ async def edit(
...
             .. versionadded:: 2.3
 
+        flags: :class:`MemberFlags`
+            The member's new flags. To know what flags are editable,
+            see `the documentation <https://discord.com/developers/docs/resources/guild#guild-member-flags>`_.
+
+            .. versionadded:: 2.8
+
+        verified: :class:`bool`
+            Whether the member bypasses guild verification requirements.
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/member.py]: nice suggestion 96a8d42e239680143494d66bf055dfae7a1206f2

File: disnake/member.py [hunk 8]
```diff
@@ -782,6 +797,8 @@ async def edit(
         +------------------------------+-------------------------------------+
         | timeout                      | :attr:`Permissions.moderate_members`|
         +------------------------------+-------------------------------------+
+        | verified                     | :attr:`Permissions.moderate_members`|
+        +------------------------------+-------------------------------------+
```
[the remark this answers is not in the document]
Comment (Victorsitou) [reply disnake/member.py]: d6367c02f2710538b477f513c900d4a6d948d6a8 & 70318fc21a126c92c478254d649bf27d10aefa06

**Comment by Victorsitou** [comment]
> It's worth mentioning that editing flags isn't documented (neither is the permission requirement); still, it works as expected and the client does it this way too.

Should we add it, though?

File: disnake/member.py [hunk 9]
```diff
@@ -816,6 +836,19 @@ async def edit(
 
             .. versionadded:: 2.3
 
+        flags: :class:`MemberFlags`
+            The member's new flags. To know what flags are editable,
+            see :ddocs:`the documentation <resources/guild#guild-member-object-guild-member-flags>`.
+
+            If parameter ``verified`` is provided, that will override the setting of :attr:`MemberFlags.bypasses_verification`.
```
Comment (shiftinv) [inline disnake/member.py:843]:
```suggestion
            If parameter ``bypasses_verification`` is provided, that will override the setting of :attr:`MemberFlags.bypasses_verification`.
```
The original comment coincided with another change, I should've suggested it like this from the start, my bad.

File: disnake/permissions.py [hunk 10]
```diff
@@ -916,7 +916,8 @@ def start_embedded_activities(self) -> int:
 
     @flag_value
     def moderate_members(self) -> int:
-        """:class:`bool`: Returns ``True`` if a user can perform limited moderation actions (timeout).
+        """:class:`bool`: Returns ``True`` if a user can perform limited moderation actions,
+        such as timeout or edit members' flags.
```
Comment (shiftinv) [inline disnake/permissions.py:920]:
```suggestion
        such as timeouts or editing members' flags.
```

File: disnake/member.py (hunk 9 above)
Comment (Victorsitou) [reply disnake/member.py:843]: ah I forgot that

**Comment by shiftinv** [comment]
> > It's worth mentioning that editing flags isn't documented (neither is the permission requirement); still, it works as expected and the client does it this way too.
> 
> Should we add it, though?

Seems fine. If anything, the edit parameter wouldn't break, it'd just be ignored; I sincerely doubt they'll change anything about it though.

**Review by shiftinv — APPROVED** [review]
lgtm, thanks!

**Pull request merged**